// game/dice-racing.js
var Class = require('lib/vendor/brejep/class');
var EntityCreator = require('game/entitycreator');
var GameManager = require('game/systems/gamemanager');
var SystemPriorities = require('game/systems/systempriorities');
var Engine = require('ash/core/engine');

module.exports = Class.extend({
        width: 0,
        height: 0,
        engine: null,
        gameState: null,
        tickProvider: null,

        constructor: function () {
          console.log('making a new game');

          this.engine = new Engine();
          var creator = new EntityCreator(this.engine, null);

          this.engine.addSystem(
            new GameManager(this.gameState, creator),
            SystemPriorities.preUpdate
          );

          /*
            var canvasContext = canvas.getContext('2d');

            this.width = canvas.width;
            this.height = canvas.height;

            this.engine = new Ash.Engine();

            this.gameState = new GameState(this.width, this.height);

            var creator = new EntityCreator(this.engine, canvasContext);

            this.engine.addSystem(
                new GameManager(this.gameState, creator),
                SystemPriorities.preUpdate
           );
            this.engine.addSystem(
                new MotionControlSystem(KeyPoll),
                SystemPriorities.update
           );
            this.engine.addSystem(
                new GunControlSystem(KeyPoll, creator),
                SystemPriorities.update
           );
            this.engine.addSystem(
                new BulletAgeSystem(creator),
                SystemPriorities.update
           );
            this.engine.addSystem(
                new MovementSystem(this.gameState),
                SystemPriorities.move
           );
            this.engine.addSystem(
                new CollisionSystem(creator),
                SystemPriorities.resolveCollisions
           );
            this.engine.addSystem(
                new RenderSystem(canvasContext),
                SystemPriorities.render
           );
            this.tickProvider = new TickProvider(stats);
          */
        },

        start: function () {
          console.log('starting the game now');

//            this.tickProvider.add(this.engine.update, this.engine);
//            this.tickProvider.start();
        }
});
